sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1909,6 +1909,12 @@ cl_int ExecCGCommand::enqueueImp() {
     if (nullptr != ExecKernel->MSyclKernel) {
       assert(ExecKernel->MSyclKernel->get_info<info::kernel::context>() ==
              Context);
+      if (Context.get_devices().size() > 1) {
```

Comment:
Could you please move this check to the sycl::program constructor?

Reply:
Done, it looks like it's better to move it to ctors of program_impl class